modules/doc/content/framework_development/code_standards.md

```diff
@@ -289,6 +291,47 @@ protected:
 
 [](---)
 
+
+## Code Commandments
+
+- Use references instead of pointers whenever possible
+    - i.e., this object lives for a shorter period of time than the object it needs to refer to does
+- Methods should return pointers to objects if returned objects are stored as pointers and references if returned objects are stored as references
+- When creating a new class:
+    - Include dependent header files in the *.C file whenever possible (i.e. the header uses only references or pointers in it's various declarations) and use forward declarations in the header file as needed
+    - One exception is when doing so would require end users to include multiple files to complete definitions of child objects (Errors typically show up as "incomplete class" or something similar during compilation)
+- Avoid using a global variable when possible.
+- Every destructor must be virtual.
+- All function definitions should be in *.C files.
+    - The only exceptions are for inline functions for speed and templates.
+- Thou shall not commit accidental insertion in a std::map by using brackets in a RHS operator unless he/she can prove it can't fail.
+- Thou shall use range-based loops or MooseIndex() based loops for iteration
+
+## Best practices
+
+-	All input parameters should always use range checked parameters where applicable
+-	Input parameters should always use MooseEnum instead of unsigned int
```

Comment:
Needs to be more specific. Maybe "input parameters accepting a finite set of permitted options" (then again the number of unsigned int values is finite as well :-D )